core/state_transition.go

```diff
@@ -253,7 +253,8 @@ func (self *StateTransition) transitionDb() (ret []byte, usedGas *big.Int, err e
 		err = nil
 	}
 
-	if vm.Debug {
+	logs := vmenv.StructLogs()
+	if len(logs) > 0 {
```

Comment:
Yes, I think it should go away. We still need `StdErrFormat` for cmd/evm, but it's fine to have it. Maybe it should be  `PrintStructLogs(io.Writer)` but that can also be done later.